minahilfaisal/edx-platform
# issue-97: removing add method deprecation warning

## Description

This pull request addresses [issue-111](https://github.com/openedx/public-engineering/issues/97) from the [openedx/public-engineering](https://github.com/openedx/public-engineering/issues) repository, removing the deprecation warning related to the use of `add` method in the [edx-platform](https://github.com/openedx/edx-platform) codebase.

The warning "[Deprecation Warning]: Using the add method to register a processor or pattern is deprecated. Use the register method instead." will no longer appear in `lms` during GitHub Actions `unit-test` runs and in the `pytest-warnings` report.

This pull request removes 12 instances of the warning across 2 files.

### Impacts
This change primarily impacts developers who work on the [Open-edX Platform](https://github.com/openedx/edx-platform) and run the `unit-test` workflow files.

### Screenshots
No UI changes are associated with this pull request.

## Supporting information

For additional context, refer to [openedx/public-engineering Issue-111](https://github.com/openedx/public-engineering/issues/97) for a more detailed discussion on the deprecation warning and the necessity of this change.

## Testing instructions

To test this change:
1. Open [this link]() for accessing the `unit-tests-gh-hosted` action/workflow for this pull request.
2. Open one of the recent runs for this workflow, such as [this run]().
3. From the summary section, the `pytest-warnings-json` file can be downloaded, such as [this one]().
4. Ensure that the deprecation warning related to using the `add` method to register a processor or pattern no longer appears in any of the `pytest-warning` files for `lms`.

## Deadline

None.

## Other information

- No dependencies on other changes.
- No special concerns or limitations.
- No database migrations are involved.
